falcon/http_error.py

```diff
@@ -165,16 +166,23 @@ def to_dict(self, obj_type=dict):
 
         return obj
 
-    def to_json(self):
-        """Return a pretty-printed JSON representation of the error.
+    def to_json(self, handler=None):
+        """Return a JSON representation of the error.
+
+        Args:
+            handler: Handler object that will be used to serialize the representation of this
+                error to JSON. When not provided a default handler using the python builtin
+                json library will be used (default ``None``).
 
         Returns:
-            str: A JSON document for the error.
+            bytes: A JSON document for the error.
 
         """
 
         obj = self.to_dict(OrderedDict)
-        return json.dumps(obj, ensure_ascii=False)
+        if not handler:
```

Comment:
not sure if it's faster or not. but it's probably easier to read

Reply:
nevermind, I prefer the if version since it's easier to see if both branches are taken. coverage does not consider the or one a branch iirc

Reply:
On consideration, I agree.
FWIW, I find the `or` pattern equally easier to read, and more performant. **However,** if a custom media handler is defined as _falsey_ for some strange reason, it would then unexpectedly fallback to the default handler, and it's a surprise we want to avoid.

Reply:
regarding performance there is actually no apparent difference. it's basically a wash.
regarding the falsey it's the same with the if, I'm just doing `not handler`, not `handler is not None` 

personally I don't think it's worth another change :)

Reply:
I.e. I prefer `if handler is None` due the falsey object issue.

Reply:
ok I guess. A lot of places do use `if obj` or `if not obj` or `obj or other` so with this in mind maybe a more general refactoring is in order?

Reply:
It depends what that `obj` is. If it is something we can control or at least expect a specific type (e.g., primitive built-ins or types defined by us), a boolean test is fine.